types/puppeteer/index.d.ts

```diff
@@ -15,6 +15,9 @@
 
 import { EventEmitter } from "events";
 import { ChildProcess } from "child_process";
+// @ts-ignore: suppress error if using an older version of puppeteer, which don't
```

Comment:
The goal here is to not force users to use a new version of puppeteer - these types are for power users, so shouldn't force everyone to update. If these are missing, the CDPSession types use a sensible fallback of method: string | number | symbol and params: any, which isn't much different from the previous.

Unfortunately there doesn't seem to be a way to disable this rule. I tried:

```
// tslint:disable-next-line ban-ts-ignore
// tslint:disable-next-line ts-ignore
```

Reply:
I'd kindly ask you to way until my puppeteer 2.0 PR lands https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40284

I'm currently working on it.

OT: This feels pretty hacky, if you really want to support TS as first class, why not move the project to TS all together?

Reply:
Porting puppeteer to TS would be a huge effort.

Yeah it's pretty hacky - I think the "don't force to update for no benefit" constraint can be avoided if we 1) wait for next puppeteer version to be released (@mathiasbynens, maybe we could do a patch release?), which will have this protocol file published and 2) tack this change onto your PR, setting the minimum version to the next (unreleased 2.0.1) version

Reply:
@connorjclark iirc the `// Type definitions for PACKAGE MAJOR.MINOR` line does not support specifying patch versions (which makes sense since the public API shouldn't change).
I think it would be reasonable to do a minor release in this case.